message/message.go

```diff
@@ -1,73 +1,84 @@
 package message
 
 import (
-	"encoding/json"
-	"errors"
-	"strings"
+	"encoding/gob"
+	"io"
+)
+
+type (
+	// Type specifies the type of a message.
+	Type int
+	// ResourceType specifies the type of a resource in a message.
+	ResourceType int
+)
+
+const (
+	// MessageRequest messages ask a peer for a resource.
+	MessageRequest Type = iota
+	// MessageResponse messages repond to a request message with an error or a resource.
+	MessageResponse
+	// MessagePush messages proactively send a resource to a peer.
+	MessagePush
 )
 
-// Message types
-// NOTE: because of the way iota works, changing the order in which the
-// following constants appear will change their values, which may affect the
-// ability of your peer to communicate with others.
 const (
-	// Send the multiaddress of a peer to another peer
-	PeerInfo = iota
-	// Request addressess of peers in the remote peer's subnet
-	RequestPeerInfo = iota
-	// Send information about a block that was just hashed
-	NewBlock = iota
-	// Request chunk of the blockchain from peer
-	RequestChunk = iota
-	// Advertise that we have a chunk of the blockchain
-	AdvertiseChunk = iota
-	// Send information about a new transaction to another peer
-	Transaction = iota
+	// ResourcePeerInfo resources contain a list of peers.
```

Comment:
Minor detail but we should update this to "... a list of peer multiaddresses" to be extra specific.